cltk/phonology/orthophonology.py

```diff
@@ -0,0 +1,477 @@
+			if type(feature_value) != feature_name:
+				raise TypeError(str(feature_value) + ' is not a ' + str(feature_name))
+
+		self.features = features
+		self.ipa = ipa
+
+	def is_vowel(self):
+		return self[Consonantal] == Consonantal.neg
+
+	def __getitem__(self, feature_name):
+		'''
+		Use dict-type syntax for accessing the values of features.
+		'''
+
+		if not issubclass(feature_name, PhonologicalFeature):
+			raise TypeError(str(feature_name) + ' is not a phonological feature')
+		return self.features.get(feature_name, None)
+
+	def __setitem__(self, feature_name, feature_value):
+		if not issubclass(feature_name, PhonologicalFeature):
+			raise TypeError(str(feature_name) + ' is not a phonological feature')
+		if type(feature_value) != feature_name:
+			raise TypeError(str(feature_value) + ' is not a ' + str(feature_name))
+		self.features[feature_name] = feature_value
+		return self
+
+	def __str__(self):
+		return  ('IPA:{0} '.format(self.ipa) if self.ipa is not None else '') + \
+		' '.join([str(v) for v in self.features.values()])
+
+	__repr__ = __str__
+
+	def is_equal(self, other):
+		return self.features == other.features
+
+	def __eq__(self, other):
+		return self.is_equal(other)
+
+
+class Consonant(AbstractPhoneme):
+	'''
+	Based on cltk.phonology.utils by @clemsciences.
+	A consonant is a phoneme that is specified for the features listed in the IPA chart for consonants:
+	Place, Manner, Voicing.  These may be read directly off the IPA chart, which also gives the IPA symbol.
+	The Consonantal feature is set to positive, and the aspirated is defaulted to negative.
+	See http://www.ipachart.com/
+	'''
+
+	def __init__(self, place, manner, voiced, ipa, geminate = Geminate.neg, aspirated = Aspirated.neg):
+		assert place is not None
+		assert manner is not None
+		assert voiced is not None
+		assert geminate is not None
+		assert ipa is not None
+
+		AbstractPhoneme.__init__(self, {
+			Consonantal : Consonantal.pos,
+			Place       : place,
+			Manner      : manner,
+			Voiced      : voiced,
+			Aspirated   : aspirated,
+			Geminate    : geminate}, 
+			ipa)
+
+	def is_more_sonorous(self, other):
+		'''
+		compare this phoneme to another for sonority.
+		Used for SSP considerations.
+		'''
+		return True if isinstance(other, Consonant) and self[Manner] > other[Manner] else False
+
+
+class Vowel(AbstractPhoneme):
+	'''
+	The representation of vowel by its features, as given in the IPA chart for vowels.
+	See http://www.ipachart.com/
+	'''
+
+	def __init__(self, height, backness, rounded, length, ipa):
+		assert height is not None
+		assert backness is not None
+		assert rounded is not None
+		assert length is not None
+		assert ipa is not None
+
+		AbstractPhoneme.__init__(self, {
+			Consonantal : Consonantal.neg,
+			Height      : height, 
+			Backness    : backness, 
+			Roundedness : rounded, 
+			Length      : length},
+			ipa)
+
+	def __add__(self, other):

```

Comment:
I see you didn't find a better solution to define diphthongs. I unfortunately don't have a better idea too.

Reply:
Yes.  I wondered about merging the feature bundles of the two vowels, but I want to consult my phonology textbook first.  Because there still are two segments, so that may be wrong.

Reply:
This is hard.  There seems to be huge variation cross-linguistically for whether bisegmental diphthongs are biphonemic or monomorphemic.  We may need a new type of phoneme here, and allow specific implementations to choose.